src/TriangleNodesProvisioner.cpp

```diff
@@ -889,6 +868,30 @@ namespace blitzdg {
     }
 
     const DGContext2D& TriangleNodesProvisioner::get_DGContext() const {
-        return DGContext;
+        DGContext2D* dgContext = new DGContext2D(
+            NumLocalPoints,
+            NumFacePoints,
+            NumElements,
+            NumFaces,
+            Filter.get(),
+            xGrid.get(),
+            yGrid.get(),
+            Fscale.get(),
+            J.get(),
+            rx.get(),
+            ry.get(),
+            sx.get(),
+            sy.get(),
+            nx.get(),
+            ny.get(),
+            Dr.get(),
+            Ds.get(),
+            Lift.get(),
+            vmapM.get(),
+            vmapP.get(),
+            BCmap.get()
+        );
+
+        return *dgContext;
     }
```

Comment:
Allocating `dgContext` with new and returning it by const reference has the potential to create a memory leak. You should instead return by value. Don't worry about the overhead of returning by value - DGContext2D is so light-weight it doesn't matter - but even if it, did the return would be either moved or elided away.

Reply:
Interesting, thanks!

Reply:
I would just write `return DGContext2D{...input params...};` obviously with the input parameters filled-in.